pkgs/tools/security/neopg/default.nix

```diff
@@ -28,10 +28,11 @@ stdenv.mkDerivation rec {
 
   doCheck = true;
   checkTarget = "test";
+  dontUseCmakeBuildDir = true;
 
-  postInstall = ''
-    mkdir -p $out/bin
-    cp src/neopg $out/bin/neopg
+  preCheck = ''
+    find | grep '\.so'
+    export LD_LIBRARY_PATH=$LD_LIBRARY_PATH:$(pwd)/3rdparty/googletest/googletest:$(pwd)/neopg
```

Comment:
Is the `:$(pwd)/neopg` required for tests to pass? It'd sound weird to me that it's necessary for tests but not for things to work, though I haven't actually tested it.

Reply:
Yeah that is the result of quite some trial-and-error. If I omit that export, I get
```
running tests
check flags: SHELL=/nix/store/yjkch3aia9ny4dq42dbcjrdwqb1y8c33-bash-4.4-p23/bin/bash VERBOSE=y test
Running tests...
/nix/store/rz7c3z0nx2wyj8a7zln46r8zc1x37srq-cmake-3.13.4/bin/ctest --force-new-ctest-process
Test project /build/source
    Start 1: NeopgTest
    Start 2: GpgErrorTest
    Start 3: AssuanTest
    Start 4: GcryptTest
1/7 Test #1: NeopgTest ........................***Failed    0.00 sec
/build/source/neopg/tests/test-libneopg: error while loading shared libraries: libneopg.so: cannot open shared object >

    Start 5: GcryptSecmemTest
2/7 Test #2: GpgErrorTest .....................***Failed    0.00 sec
/build/source/legacy/gpg-error-test: error while loading shared libraries: libgtest_main.so: cannot open shared object>

    Start 6: KsbaTest
3/7 Test #3: AssuanTest .......................***Failed    0.00 sec
/build/source/legacy/assuan-test: error while loading shared libraries: libgtest_main.so: cannot open shared object fi>

    Start 7: NeopgToolTest
4/7 Test #4: GcryptTest .......................***Failed    0.00 sec
/build/source/legacy/gcrypt-test: error while loading shared libraries: libgtest_main.so: cannot open shared object fi>

5/7 Test #5: GcryptSecmemTest .................***Failed    0.00 sec
/build/source/legacy/gcrypt-secmem-test: error while loading shared libraries: libgtest_main.so: cannot open shared ob>

6/7 Test #6: KsbaTest .........................***Failed    0.00 sec
/build/source/legacy/ksba-test: error while loading shared libraries: libgtest_main.so: cannot open shared object file>

7/7 Test #7: NeopgToolTest ....................***Failed    0.00 sec
/build/source/neopg-tool/tests/test-neopg: error while loading shared libraries: libgtest_main.so: cannot open shared >


0% tests passed, 7 tests failed out of 7

Total Test time (real) =   0.01 sec

The following tests FAILED:
          1 - NeopgTest (Failed)
          2 - GpgErrorTest (Failed)
          3 - AssuanTest (Failed)
          4 - GcryptTest (Failed)
          5 - GcryptSecmemTest (Failed)
          6 - KsbaTest (Failed)
          7 - NeopgToolTest (Failed)
Errors while running CTest
make: *** [Makefile:130: test] Error 8
builder for '/nix/store/gb1p2kiqzh6k9zbvi2yjkkijf4bqd862-neopg-0.0.6pre2018-11-10.drv' failed with exit code 2
```
So the `.so` files are not found for some reason. Most annoyingly I even can't reproduce that behaviour by manully building the package in a nix-shell.

As the resulting binary `neopg` executes just fine, I believe that everything is alright.